dev/webpack.config.js

```diff
@@ -0,0 +1,161 @@
+/* eslint import/no-commonjs: 0 */
+const path = require('path');
+const without = require('lodash/without');
+
+const webpack = require('webpack');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const HtmlWebpackPluging = require('html-webpack-plugin');
+const BabiliPlugin = require('babili-webpack-plugin');
+const AutoDllPlugin = require('autodll-webpack-plugin');
+const HappyPack = require('happypack');
+
+const __DEV__ = process.env.NODE_ENV === 'development';
+const clean = arr => arr.filter(item => item !== false);
+
+module.exports = {
+  devtool: __DEV__ ? 'cheap-eval-source-map' : 'cheap-module-source-map',
+
+  entry: __DEV__
+    ? {
+      bundle: [
+        'webpack-dev-server/client?http://localhost:8080',
+        'webpack/hot/only-dev-server',
+        './dev/app',
+      ],
+      instantsearch: './index.js',
+    }
+    : {
+      bundle: './dev/app',
+    },
+
+  output: {
+    path: path.join(__dirname, 'dist'),
+    filename: '[name].[hash].js',
+    library: '[name]',
+    libraryTarget: 'umd',
+  },
+
+  module: {
+    rules: clean([
```

Comment:
We can rename `clean()` to something else, I wasn't inspired when doing it. 

I really like having this syntax: `__DEV__ && ...` it's so explicit that plugin or option is meant for development environment or not. 

We should not bother on how clean is the code for configurations files but on how simple is it to read and understand.